uaa requires the disabling of postgres TLS to be used correctly

We had to add custom ops file for UAA to be able to connect to the same postgres instance that is allocated for Concourse via `cluster`.

```yaml
- type: replace
  path: /instance_groups/name=web/jobs/name=uaa/properties/uaadb?/tls
  value: disabled
```

Options:
1. Fix UAA ops file so it uses TLS.
2. Always disable it.

Just for reference, without this change, the UAA has a stacktrace in its log.

```
Caused by: org.postgresql.util.PSQLException: The server does not support SSL.
        at org.postgresql.core.v3.ConnectionFactoryImpl.enableSSL(ConnectionFactoryImpl.java:431) ~[postgresql-42.2.8.jar:42.2.8]
        at org.postgresql.core.v3.ConnectionFactoryImpl.tryConnect(ConnectionFactoryImpl.java:94) ~[postgresql-42.2.8.jar:42.2.8]
        at org.postgresql.core.v3.ConnectionFactoryImpl.openConnectionImpl(ConnectionFactoryImpl.java:192) ~[postgresql-42.2.8.jar:42.2.8]
        at org.postgresql.core.ConnectionFactory.openConnection(ConnectionFactory.java:49) ~[postgresql-42.2.8.jar:42.2.8]
        at org.postgresql.jdbc.PgConnection.<init>(PgConnection.java:195) ~[postgresql-42.2.8.jar:42.2.8]
        at org.postgresql.Driver.makeConnection(Driver.java:458) ~[postgresql-42.2.8.jar:42.2.8]
        at org.postgresql.Driver.connect(Driver.java:260) ~[postgresql-42.2.8.jar:42.2.8]
        at org.apache.tomcat.jdbc.pool.PooledConnection.connectUsingDriver(PooledConnection.java:319) ~[tomcat-jdbc-9.0.27.jar:?]
        at org.apache.tomcat.jdbc.pool.PooledConnection.connect(PooledConnection.java:212) ~[tomcat-jdbc-9.0.27.jar:?]
        at org.apache.tomcat.jdbc.pool.ConnectionPool.createConnection(ConnectionPool.java:744) ~[tomcat-jdbc-9.0.27.jar:?]
        at org.apache.tomcat.jdbc.pool.ConnectionPool.borrowConnection(ConnectionPool.java:676) ~[tomcat-jdbc-9.0.27.jar:?]
        at org.apache.tomcat.jdbc.pool.ConnectionPool.init(ConnectionPool.java:483) ~[tomcat-jdbc-9.0.27.jar:?]
        at org.apache.tomcat.jdbc.pool.ConnectionPool.<init>(ConnectionPool.java:154) ~[tomcat-jdbc-9.0.27.jar:?]
        at org.apache.tomcat.jdbc.pool.DataSourceProxy.pCreatePool(DataSourceProxy.java:118) ~[tomcat-jdbc-9.0.27.jar:?]
        at org.apache.tomcat.jdbc.pool.DataSourceProxy.createPool(DataSourceProxy.java:107) ~[tomcat-jdbc-9.0.27.jar:?]
        at org.apache.tomcat.jdbc.pool.DataSourceProxy.getConnection(DataSourceProxy.java:131) ~[tomcat-jdbc-9.0.27.jar:?]
        at org.flywaydb.core.internal.jdbc.JdbcUtils.openConnection(JdbcUtils.java:56) ~[flyway-core-5.2.4.jar:?]
        at org.flywaydb.core.internal.database.DatabaseFactory.createDatabase(DatabaseFactory.java:72) ~[flyway-core-5.2.4.jar:?]
        at org.flywaydb.core.Flyway.execute(Flyway.java:1670) ~[flyway-core-5.2.4.jar:?]
        at org.flywaydb.core.Flyway.repair(Flyway.java:1566) ~[flyway-core-5.2.4.jar:?]
        at org.cloudfoundry.identity.uaa.db.beans.FlywayConfiguration.flyway(FlywayConfiguration.java:43) ~[cloudfoundry-identity-server-74.9.0.jar:?]
        at org.cloudfoundry.identity.uaa.db.beans.FlywayConfiguration$$EnhancerBySpringCGLIB$$f525c96e.CGLIB$flyway$0(<generated>) ~[cloudfoundry-identity-s
erver-74.9.0.jar:?]
        at org.cloudfoundry.identity.uaa.db.beans.FlywayConfiguration$$EnhancerBySpringCGLIB$$f525c96e$$FastClassBySpringCGLIB$$776d01f8.invoke(<generated>)
 ~[cloudfoundry-identity-server-74.9.0.jar:?]
        at org.springframework.cglib.proxy.MethodProxy.invokeSuper(MethodProxy.java:244) ~[spring-core-5.2.1.RELEASE.jar:5.2.1.RELEASE]
        at org.springframework.context.annotation.ConfigurationClassEnhancer$BeanMethodInterceptor.intercept(ConfigurationClassEnhancer.java:363) ~[spring-c
ontext-5.2.1.RELEASE.jar:5.2.1.RELEASE]
```

--- a/cluster/operations/secure-internal-postgres-bbr.yml
+++ b/cluster/operations/secure-internal-postgres-bbr.yml
@@ -1,0 +1,17 @@
+# requires the secure-internal-postgres.yml op file
+- type: replace
+  path: /variables/-
+  value:
+    name: postgres_client_tls
+    type: certificate
+    options:
+      ca: postgres_ca
+      common_name: concourse
+      extended_key_usage:
+      - client_auth
+
+- type: replace
+  path: /instance_groups/name=web/jobs/name=web/properties/postgresql?/client_cert
+  value:
+    private_key: ((postgres_client_tls.private_key))
+    certificate: ((postgres_client_tls.certificate))

--- a/cluster/operations/secure-internal-postgres-credhub.yml
+++ b/cluster/operations/secure-internal-postgres-credhub.yml
@@ -1,6 +1,6 @@
 # requires the secure-internal-postgres.yml op file
 - type: replace
-  path: /instance_groups/name=web/jobs/name=credhub?/properties?/credhub?/data_storage?/tls_ca?
+  path: /instance_groups/name=web/jobs/name=credhub?/properties/credhub/data_storage/tls_ca?
   value: ((postgres_ca.certificate))
 
 - type: replace

--- a/cluster/operations/secure-internal-postgres-credhub.yml
+++ b/cluster/operations/secure-internal-postgres-credhub.yml
@@ -1,0 +1,8 @@
+# requires the secure-internal-postgres.yml op file
+- type: replace
+  path: /instance_groups/name=web/jobs/name=credhub?/properties?/credhub?/data_storage?/tls_ca?
+  value: ((postgres_ca.certificate))
+
+- type: replace
+  path: /instance_groups/name=web/jobs/name=credhub?/properties/credhub/data_storage/require_tls?
+  value: true

--- a/cluster/operations/secure-internal-postgres-uaa.yml
+++ b/cluster/operations/secure-internal-postgres-uaa.yml
@@ -1,0 +1,8 @@
+# requires the secure-internal-postgres.yml op file
+- type: replace
+  path: /instance_groups/name=web/jobs/name=uaa?/properties/uaa/ca_certs?
+  value: [((postgres_ca.certificate))]
+
+- type: replace
+  path: /instance_groups/name=web/jobs/name=uaa?/properties/uaadb/tls?
+  value: enabled

--- a/cluster/operations/secure-internal-postgres.yml
+++ b/cluster/operations/secure-internal-postgres.yml
@@ -1,0 +1,43 @@
+- type: replace
+  path: /variables/-
+  value:
+    name: postgres_ca
+    type: certificate
+    options:
+      is_ca: true
+      common_name: postgresCA
+
+- type: replace
+  path: /variables/-
+  value:
+    name: postgres_tls
+    type: certificate
+    options:
+      ca: postgres_ca
+      extended_key_usage:
+      - server_auth
+    consumes:
+      alternative_name: { from: concourse-postgres-address }
+      common_name: { from: concourse-postgres-address }
+
+- type: replace
+  path: /instance_groups/name=web/jobs/name=web/properties/postgresql/ca_cert?
+  value: ((postgres_ca))
+
+- type: replace
+  path: /instance_groups/name=web/jobs/name=web/properties/postgresql/sslmode?
+  value: verify-ca
+
+- type: replace
+  path: /instance_groups/name=db/jobs/name=postgres/provides?/postgres-address
+  value: {as: concourse-postgres-address}
+
+- type: replace
+  path: /instance_groups/name=db/jobs/name=postgres/custom_provider_definitions?/-
+  value: {name: postgres-address, type: address}
+
+- type: replace
+  path: /instance_groups/name=db/jobs/name=postgres/properties/databases?/tls
+  value:
+    private_key: ((postgres_tls.private_key))
+    certificate: ((postgres_tls.certificate))

